stac_fastapi/elasticsearch/core.py

```diff
@@ -204,12 +213,13 @@ def get_search(
             response_item = serializers.ItemSerializer.db_to_stac(item, base_url)
             response.append(response_item)
 
-        links = generate_pagination_links(kwargs['request'])
+        links = generate_pagination_links(kwargs['request'], items.count(), limit)
 
         return stac_types.ItemCollection(
             type='FeatureCollection',
             features=response,
-            links=links
+            links=links,
+            context=context,
```

Comment:
likewise with this, the other hook is to modify the response so we should probably move to use:

```python
response = stac_types.ItemCollection(
    ...
)

if self.extension_is_enabled('ContextExtension'):
    context = {
                'returned':limit if page*limit <= matched-1 else (matched-1)-(page-1)*limit,
                'limit':limit,
                'matched':matched,
            }    
    response['context'] = context

return response
```
**Untested**